Tasks/replace.js

```diff
@@ -1,22 +1,18 @@
+'use strict';
 // Replace substring with newstr
 
-Replace = (str, substr, newstr) => {
-  if (substr === '') {
-    return str;
-  } else {
-    src = str;
-    res = '';
-    do {
-      const _index = src.indexOf(substr);
-      if (_index === -1) {
-        return res + src;
-      } else {
-        const start = src.substring(0, _index);
-        src = src.substring(_index + substr.length, src.length);
-        res += start + newstr;
-      }
-    } while (true);
+const replace = (str, substr, newstr) => {
+  let res = '';
+  const substrLen = substr.length;
+  if (substrLen) {
+    let substrIndex = str.indexOf(substr);
+    while (substrIndex !== -1){
+      res += str.substring(0, substrIndex) + newstr;
+      str = str.substring(substrIndex + substrLen);
```

Comment:
Try not to change input arguments, it's bad practice.
In general, there is a ready-made [string method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace) for this task.
`String.prototype.replace()`